packages/next-swc/crates/core/src/styled_jsx/transform_css.rs

```diff
@@ -162,88 +156,177 @@ struct Namespacer {
+            let (name, args) = match selector {
+                SubclassSelector::PseudoClass(PseudoClassSelector { name, children, .. }) => {
+                    match children {
+                        Some(PseudoSelectorChildren::Nth(_)) => todo!("nth"),
+                        Some(PseudoSelectorChildren::Tokens(v)) => (name, v),
+                        None => (name, &empty_tokens),
+                    }
+                }
+                SubclassSelector::PseudoElement(PseudoElementSelector {
+                    name, children, ..
+                }) => match children {
+                    Some(children) => (name, children),
+                    None => (name, &empty_tokens),
+                },
+                _ => continue,
+            };
+
+            // One off global selector
+            if &name.value == "global" {
+                let block_tokens = get_block_tokens(&args);
+                let mut front_tokens = get_front_selector_tokens(&args);
+                let mut args = args.clone();
+                front_tokens.extend(args.tokens);
+                front_tokens.extend(block_tokens);
+                args.tokens = front_tokens;
+
+                let complex_selectors = panic::catch_unwind(|| {
+                    let x: ComplexSelector = parse_tokens(
+                        &args,
+                        ParserConfig {
+                            parse_values: false,
+                            allow_wrong_line_comments: true,
+                        },
+                        // TODO(kdy1): We might be able to report syntax errors.
+                        &mut vec![],
+                    )
+                    .unwrap();
+                    return x;
+                });
+
+                return match complex_selectors {
+                    Ok(complex_selectors) => {
+                        let mut v = complex_selectors.children[1..]
+                            .iter()
+                            .cloned()
+                            .collect::<Vec<_>>();
+
+                        match v[0] {
+                            ComplexSelectorChildren::Combinator(Combinator {
+                                value: CombinatorValue::Descendant,
+                                ..
+                            }) => {
+                                v.remove(0);
                             }
+                            _ => {}
+                        }
 
-                            if node.combinator.is_some() && v[0].combinator.is_some() {
-                                bail!("Failed to transform one off global selector");
-                            } else if node.combinator.is_some() {
-                                v[0].combinator = node.combinator;
+                        if v.is_empty() {
+                            bail!("Failed to transform one off global selector");
+                        }
+
+                        trace!("Combinator: {:?}", combinator);
```

Comment:
cc @padmaia 
This line is also removable. I used it to debug `styled-jsx`, but I'm not sure if it's better to remove it.
Also completely noop on release builds.

Reply:
Same answer as the `debug` comment